### ISSUE TITLE
Webpack configs should be merged using webpack-merge

### ISSUE THREAD

**Issue by mikiest** [flexn-io/renative#669]
**Describe the bug**
When using `platforms.web.webpackConfig.extend` to extend some of the webpack configs for `web`, the webpack configs are currently merged using a spread operator `...extend`. This means any object we specify in `platforms.web.webpackConfig.extend`, for example `resolve`, will override the default equivalent entirely instead of being merged with it.

**To Reproduce**
1. In `renative.json`, add the following:
```json
"platforms": {
	"web": {
		"webpackConfig": {
			"extend": {
				"resolve": {
					"alias": {
						  "my-module": "some_path"
					}
				}
			}
		}
	}
}
```
2. The build won't succeed, for example most `react-native` modules will not be found anymore, TypeScript file extensions would not necessarily be included, etc.
This is a result of `extensions` and other aliases being removed since spreading the content of the `extend` object above will override entirely its properties and not deep merge them.

**Expected behavior**
All webpack configs are always merged using [webpack-merge](https://www.npmjs.com/package/webpack-merge) so that configs can be extended and not overriden.

**Screenshots or copy&paste**
This is where I believe using `webpack-merge` would be better, for example, in `platformBuilds/<projectName>_web/webpack.config.dev.js`:
```js
module.exports = {
    entry: C.entry,
    devServer: C.devServer,
    output: C.output,
    module: {
        rules: [C.Rules.babel, C.Rules.css, C.Rules.image, C.Rules.fonts, C.Rules.sourcemap],
    },
    plugins,
    resolve: {
        symlinks: false,
        extensions: C.extensions,
        alias: C.aliases,
    },
    ...config.extend || {}	// this will override `resolve` entirely if `config.extend` contains the same property
};
```

**Desktop:**
- OS: OSX
- Node Version: v10.16.3
- RNV Version: 0.31.3

**Comment by pauliusguzas** [comment]
@mikiest thanks for reporting this issue. Is this still reproduced on latest rnv version?

**Comment by pauliusguzas** [comment]
@mikiest should be fixed with 1.0.0-rc.12
